bootstrap/kubeadm/controllers/kubeadmconfig_controller.go

```diff
@@ -435,9 +436,32 @@ func (r *KubeadmConfigReconciler) handleClusterNotInitialized(ctx context.Contex
 		InitConfiguration:    initdata,
 		ClusterConfiguration: clusterdata,
 		Certificates:         certificates,
-	})
+	}
+
+	var cloudInitData []byte
+
+	switch scope.Config.Spec.Format {
+	case bootstrapv1.Ignition:
+		ign, warnings, err := ignition.NewInitControlPlane(&ignition.ControlPlaneInput{
+			ControlPlaneInput: controlPlaneInput,
+			Ignition:          scope.Config.Spec.Ignition,
+		})
+		if err != nil {
+			scope.Error(err, "Failed to Ignition user data for bootstrap control plane")
+			return ctrl.Result{}, err
+		}
+
+		if warnings != "" {
+			conditions.MarkFalse(scope.Config, bootstrapv1.IgnitionGeneratedCondition, bootstrapv1.IgnitionHasWarningsReason, clusterv1.ConditionSeverityWarning, warnings)
```

Comment:
if IgnitionGeneratedCondition status=false, does it literally mean that ignition is not generated? maybe change the condition name?

Reply:
>if IgnitionGeneratedCondition status=false, does it literally mean that ignition is not generated?

If there are some warnings, Ignition is still generated.

>maybe change the condition name?



I found it difficult to come up with reasonable names here. Do you have some better specific suggestions?

Reply:
The condition and warnings are not part of this PR anymore due to complexity they bring. They will also be possibly moved to webhook itself, so user can see them directly. I'll mark this as resolved.